README.md

```diff
@@ -1,449 +1,62 @@
-
-  * ListObjectsPages - Same as the base API operation, but uses a callback to
-    automatically handle pagination of the API's response.
-
-  * ListObjectsWithContext - Same as base API operation, but adds support for
-    the Context pattern. This is helpful for controlling the canceling of in
-    flight requests. See the Go standard library context package for more
-    information. This method also takes request package's Option functional
-    options as the variadic argument for modifying how the request will be
-    made, or extracting information from the raw HTTP response.
-
-  * ListObjectsPagesWithContext - same as ListObjectsPages, but adds support for
-    the Context pattern. Similar to ListObjectsWithContext this method also
-    takes the request package's Option function option types as the variadic
-    argument.
+and wrap it
 
-In addition to the API operations the SDK also includes several higher level
-methods that abstract checking for and waiting for an AWS resource to be in
-a desired state. In this list we'll use WaitUntilBucketExists to demonstrate
-the different forms of waiters.
+``` go
+package main
 
-  * WaitUntilBucketExists. - Method to make API request to query an AWS service for
-    a resource's state. Will return successfully when that state is accomplished.
+import (
+	"fmt"
+	"log"
 
-  * WaitUntilBucketExistsWithContext - Same as WaitUntilBucketExists, but adds
-    support for the Context pattern. In addition these methods take request
-    package's WaiterOptions to configure the waiter, and how underlying request
-    will be made by the SDK.
+	"github.com/Clever/aws-sdk-go-counter/counter/s3counter"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
 
-The API method will document which error codes the service might return for
-the operation. These errors will also be available as const strings prefixed
-with "ErrCode" in the service client's package. If there are no errors listed
-in the API's SDK documentation you'll need to consult the AWS service's API
-documentation for the errors that could be returned.
-
-```go
-  ctx := context.Background()
-
-  result, err := svc.GetObjectWithContext(ctx, &s3.GetObjectInput{
-      Bucket: aws.String("my-bucket"),
-      Key: aws.String("my-key"),
-  })
-  if err != nil {
-      // Cast err to awserr.Error to handle specific error codes.
-      aerr, ok := err.(awserr.Error)
-      if ok && aerr.Code() == s3.ErrCodeNoSuchKey {
-          // Specific error code handling
-      }
-      return err
-  }
-
-  // Make sure to close the body when done with it for S3 GetObject APIs or
-  // will leak connections.
-  defer result.Body.Close()
-
-  fmt.Println("Object Size:", aws.StringValue(result.ContentLength))
+func main() {
+	sess := session.New()
+	svc := s3counter.New(s3.New(sess))
+	_, err := svc.ListBuckets(nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("%v\n", svc.Counters())
```

Comment:
It might be worthwhile to show how one might get these counters into our logging pipeline, perhaps by referencing one of the workflow manager changes.

Reply:
c5c6416